packages/panna-sdk/src/react/hooks/use-onramp-session-status.ts

```diff
@@ -0,0 +1,99 @@
+import { useQuery, UseQueryOptions } from '@tanstack/react-query';
+import { getValidSiweAuthToken } from 'src/core/auth';
+import { type SessionStatusResult } from 'src/core/onramp/onramp-money';
+import { pannaApiService } from 'src/core/util/api-service';
+import {
+  createDefaultRetryFn,
+  DEFAULT_RETRY_DELAY,
+  DEFAULT_STALE_TIME
+} from './constants';
+import { usePanna } from './use-panna';
+
+/**
+ * Parameters for useOnrampSessionStatus hook
+ */
+type UseOnrampSessionStatusParams = {
+  sessionId: string;
+};
+
+/**
+ * Hook to retrieve and poll onramp.money session status
+ *
+ * This hook automatically polls the session status every 5 seconds when the status
+ * is non-terminal (created or pending). Polling stops when the session reaches a
+ * terminal state (completed, failed, cancelled, or expired).
+ *
+ * The hook automatically retrieves and includes the JWT auth token if available.
```

Comment:
We need to ensure that JWT token if unavailable must be generated with the SIWE auth process.